ec2/InstanceType.dhall

```diff
@@ -1267,40 +1137,102 @@ let alternatives =
                 , r5a-4xlarge =
                   [ InstanceType.r5-4xlarge, InstanceType.r5n-4xlarge ]
                 , r5n-large =
-                          if _params.permitWorseNetwork
-
+                    if    _params.permitWorseNetwork
                     then  [ InstanceType.r5-large, InstanceType.r5a-large ]
-
                     else  [] : List InstanceType
                 , r5n-xlarge =
-                          if _params.permitWorseNetwork
-
+                    if    _params.permitWorseNetwork
                     then  [ InstanceType.r5-xlarge, InstanceType.r5a-xlarge ]
-
                     else  [] : List InstanceType
                 , r5n-2xlarge =
-                          if _params.permitWorseNetwork
-
+                    if    _params.permitWorseNetwork
                     then  [ InstanceType.r5-2xlarge, InstanceType.r5a-2xlarge ]
-
                     else  [] : List InstanceType
                 , r5n-4xlarge =
-                          if _params.permitWorseNetwork
-
+                    if    _params.permitWorseNetwork
                     then  [ InstanceType.r5-4xlarge, InstanceType.r5a-4xlarge ]
-
+                    else  [] : List InstanceType
+                , t3-nano =
+                    if    _params.permitCrossInstanceFamily
+                    then  [ InstanceType.c5-large
+                          , InstanceType.c5d-large
+                          , InstanceType.m5-large
+                          , InstanceType.m5d-large
```

Comment:
There are also `m5n` and `m5dn` which are acceptable substitutes (yes, sometimes they are cheaper on the spot market than an `m5`, it was an interesting surprise we discovered when we first wrote this).

You should also check for the `_params.permitLowerClassCPU` and, if enabled, permit the following families: `m4`, `m5a`, `m5ad`, and `r5a`.

Reply:
I'll add m5n, m5dn. As for the lower class cpu --> for me, I need it strict and can't allow this. If we'll need it, I'll add it in later phase